graphql_compiler/compiler/emit_sql.py

```diff
@@ -477,58 +498,64 @@ def _get_mssql_xml_path_column(
                             used in subquery.
 
         Returns:
             Selectable for XML PATH aggregation subquery
         """
-        # use join info tuple for most recent traversal to set WHERE clause for XML PATH subquery
+        if self._output_vertex_alias is None:
+            raise AssertionError(
+                "Attempted to aggregate with XML PATH before an _output_vertex_alias had been "
+                "found."
+            )
+        # Use join info tuple for most recent traversal to set WHERE clause for XML PATH subquery.
         edge, from_alias, to_alias = last_traversal
 
         return _get_xml_path_clause(
             self._output_vertex_alias.c[fold_output_field],
             (from_alias.c[edge.from_column] == to_alias.c[edge.to_column]),
         ).label(intermediate_fold_output_name)
 
-    def _get_fold_output_column_clause(self, fold_output_field):
+    def _get_fold_output_column_clause(self, fold_output_field: str) -> Label:
         """Get the SQLAlchemy column expression corresponding to the fold output field."""
         if fold_output_field == COUNT_META_FIELD_NAME:
             return sqlalchemy.func.coalesce(
                 sqlalchemy.func.count(), sqlalchemy.literal_column("0")
             ).label(FOLD_OUTPUT_FORMAT_STRING.format(COUNT_META_FIELD_NAME))
         else:
-            # force column to have explicit label as opposed to anon_label
+            # Force column to have explicit label as opposed to anon_label.
             intermediate_fold_output_name = FOLD_OUTPUT_FORMAT_STRING.format(fold_output_field)
-            # add array aggregated output column to self._outputs
-            # add aggregated output column to self._outputs
+            # Add aggregated output column to self._outputs.
             if isinstance(self._dialect, MSDialect):
-                # MSSQL uses XML PATH aggregation
+                # MSSQL uses XML PATH aggregation.
                 return self._get_mssql_xml_path_column(
                     intermediate_fold_output_name,
                     fold_output_field,
                     # output is last vertex traversed
                     self._traversal_descriptors[-1],
                 )
             elif isinstance(self._dialect, PGDialect):
-                # PostgreSQL uses ARRAY_AGG
+                # PostgreSQL uses ARRAY_AGG.
                 return self._get_array_agg_column(intermediate_fold_output_name, fold_output_field)
             else:
                 raise NotImplementedError(
                     "Fold only supported for MSSQL and PostgreSQL, "
-                    "dialect set to {}".format(self._dialect.name)
+                    f"dialect set to {self._dialect.name}."
                 )
 
-        # We should have either triggered a not implemented error, or returned earlier
         raise AssertionError(
             "Reached end of function without returning a value, this code should be unreachable."
```

Comment:
I know this isn't part of the review, so feel free to ignore — just wanted to mention another place where outputting more info would make debugging the resulting error much less frustrating.